src/main/scala/path/Relationship.scala

```diff
@@ -23,29 +24,33 @@ sealed trait Relationship
 
   protected def makeRelationshipString(
     identifiableMap: IdentifiableMap,
-    kinds: Seq[String],
+    kinds: Seq[Label],
     rest: String = ""
   ): String =
-    s"[${getIdentifierOrEmptyString(identifiableMap, this)}${kindsToQuery(kinds)}$rest]"
+    s"[${getIdentifierOrEmptyString(identifiableMap, this)}${Relationship.kindsToQuery(kinds)}$rest]"
 
-  private def kindsToQuery(kinds: Seq[String]): String =
+}
+
+object Relationship {
+
+  def kindsToQuery(kinds: Seq[Label]): String =
     if (kinds.isEmpty) ""
     else ":" + (kinds map escape mkString "|")
 
-  private def escape(kind: String): String =
-    if (kind contains '`') throw new CharacterNotAllowedInLabel('`', kind)
-    else if (kind contains ' ') s"`$kind`"
+  def escape(kind: Label): String =
```

Comment:
the `escape` function can now move to `Label` as `escapedName`. then in `kindsToQuery` it becomes `kinds map (_.escapedName) mkString "|")`